manager/controlapi/extension.go

```diff
@@ -13,7 +18,27 @@ import (
 //   or fails validation.
 // - Returns an error if the creation fails.
 func (s *Server) CreateExtension(ctx context.Context, request *api.CreateExtensionRequest) (*api.CreateExtensionResponse, error) {
-	return nil, errors.New("not implemented")
+	if request.Extension == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "extension must be provided")
+	}
+
+	err := s.store.Update(func(tx store.Tx) error {
+		return store.CreateExtension(tx, request.Extension)
+	})
+
+	switch err {
+	case store.ErrNameConflict:
+		return nil, status.Errorf(codes.AlreadyExists, "extension %s already exists", request.Extension.Annotations.Name)
+	case nil:
+		log.G(ctx).WithFields(logrus.Fields{
+			"extension.Name": request.Extension.Annotations.Name,
+			"method":         "CreateExtension",
+		}).Debugf("extension created")
+
+		return &api.CreateExtensionResponse{}, nil
```

Comment:
I didn't notice this when reviewing the last PR, with the API, but the `CreateExtensionResponse` and `CreateResourceResponse` should contain the Extension or Resource that has been created, in keeping with the existing create APIs.